lib/BaseDownloadController.js

```diff
@@ -77,57 +81,141 @@ module.exports = class BaseDownloadController extends CompositeController {
         }
       }
 
-      try {
-        let res = await this.download(reqOpt);
-        if (res.toJSON instanceof Function) {
-          res = res.toJSON();
-        }
-        if (res.statusCode >= 200 && res.statusCode < 300) {
-          this.log.debug(`Download ${res.statusCode} ${url}`);
-          file = yaml.loadAll(res.body);
-          if (Array.isArray(file) && file.length == 1) { file = file[0]; }
-
-          // TODO if last-modified doesnt exist try etag
-          // use with request at .headers.If-None-Match
-          let resLM = objectPath.get(res, 'headers.last-modified');
-          if (resLM) {
-            await fs.outputJson(fileCachePath, file);
-            newLastModifiedArray[i] = { hash: requestHash, url: url, 'last-modified': resLM };
+      const git = objectPath.get(req, 'options.git');
+      let host;
+      let repo;
+      let branch;
+      let requrl;
+      let files;
+      let path;
+      let fileExt;
+      let filename;
+      if (git) {
+        const parse = gh(git.repo);
```

Comment:
Will the `gh` module work with any git repo?

Reply:
it at least works for github and gitlab, seems to work for bitbucket too but haven't tested much, not sure about others.